aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +80,101 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int64_t ax = A.size(0);
+  int64_t ay = A.size(1);
+  int info, res;
+  at::optional<Tensor> temp_sol;
+  at::optional<Tensor> temp_lu;
+
+  /* Init to column major format. `sol` and `lu` need to be contiguous
+   * since we pass sol.data() and lu.data() to Lapack */
+
+  bool tc = isTransposeContiguous(sol);
+  if (tc) {
+    // if transpose of output tensor is contiguous, use it
+    sol = sol.t();
+  } else if (sol.dim() == 2 &&
+             sol.size(0) == bx &&
+             sol.size(1) == by) {
+    temp_sol = self.type().tensor();
+  }
+
+  if (temp_sol) {
+    // guard against users passing correct size but incorrect stride
+    temp_sol.value().resize_({by, bx});
+    temp_sol.value().copy_(self.view({bx, by}).t());
+  } else {
+    sol.resize_({by, bx});
+    if (&self == &sol) {
+      // note: not a comprehensive check. Will fail for views
+      if (!tc) {
+        sol.copy_(self.view({bx, by}).t().clone());
+      }
+    } else {
+      sol.copy_(self.view({bx, by}).t());
+    }
+  }
+
+  tc = isTransposeContiguous(lu);
+  if (tc) {
+    lu = lu.t();
+  } else if (lu.dim() == 2 &&
+             lu.size(0) == ax &&
+             lu.size(1) == ay) {
+    temp_lu = self.type().tensor();
+  }
+
+  if (temp_lu) {
+    temp_lu.value().resize_({ay, ax});
+    temp_lu.value().copy_(A.t());
+  } else {
+    lu.resize_({ay, ax});
+    if (&A == &lu) {
+      if (!tc) {
+        lu.copy_(A.t().clone());
+      }
+    } else {
+      lu.copy_(A.t());
+    }
+  }
+
+  AT_DISPATCH_FLOATING_TYPES(self.type(), "gesv", [&]{
+    auto A_ptr = temp_lu ? temp_lu.value().data<scalar_t>()
+                         : lu.data<scalar_t>();
+    auto b_ptr = temp_sol ? temp_sol.value().data<scalar_t>()
+                          : sol.data<scalar_t>();
+    auto ipiv = at::empty({bx}, sol.options().dtype(kInt));
+    res = lapackGesv<scalar_t>(bx, by, A_ptr, bx, ipiv.data<int>(),
+                               b_ptr, bx, &info);
+  });
+
+  if (temp_sol) {
+    sol.copy_(temp_sol.value().t_());
+  } else {
+    sol = sol.t();
+  }
+
+  if (temp_lu) {
+    lu.copy_(temp_lu.value().t_());
+  } else {
+    lu = lu.t();
+  }
+
+  if (res == -1) {
+    AT_ERROR("gesv only takes float or double Tensors");
```

Comment:
This should already error at `AT_DISPATCH_FLOATING_TYPES`.... please revert this.